### ISSUE TITLE
[ban-ts-comment] multiline @ts-ignore and @ts-expect-error not supported

### ISSUE THREAD

**Issue by dimitropoulos** [typescript-eslint/typescript-eslint#2617]
The [regex we use](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/ban-ts-comment.ts#L101) for detecting TypeScript directives was originally adapted [from tslint](https://github.com/palantir/tslint/blob/master/src/rules/banTsIgnoreRule.ts#L48).  Since then (I think??) TypeScript has added multiline comment support for directives.

You can see TypeScript's own regexs [here](https://github.com/microsoft/TypeScript/blob/master/src/compiler/scanner.ts#L281-L289) (they have one for single line and another for multiline) as well as some useful test cases [here](https://github.com/microsoft/TypeScript/blob/master/tests/cases/conformance/directives/multiline.tsx).

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-ts-comment": "error"
  }
}
```

```TS

if (false) {
  // @ts-expect-error
  console.log('this errors (appropriately) because a description is missing');
}

if (false) {
  // @ts-expect-error explanation: ziltoid is omniscient
  console.log('this does not error (appropriately) because a description is present');
}

if (false) {
  /* @ts-expect-error */
  console.log('this one does not error, although it should');
}
```

**Expected Result**

```ts
if (false) {
  /* @ts-expect-error */
  console.log('this one does not error, although it should');
}
```

should return an error.

**Actual Result**

no error is returned.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.1` |
| `@typescript-eslint/parser`        | `4.1.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `12.6.0` |

**Comment by dimitropoulos** [comment]
I'll note that while I'd be happy to fix this, this would be a great first contribution from someone because the typescript tests (linked above) and the real-life regex used by typescript (also linked above) are already written.  We simply need to update our regex here to match (or, add two like typescript did, if so, and check both) and add some new tests (which, again, can simply mirror theirs).

**Comment by bradzacher** [comment]
prior art for a similar fix for `prefer-ts-expect-error`: https://github.com/typescript-eslint/typescript-eslint/pull/2541

**Comment by dimitropoulos** [comment]
cool, thanks.  I'll make a reminder to come check back on this in a little while.

In the meantime if anyone sees this and wants to give it a try I'd be happy to help however I can - pair programming included.

**Comment by parynaz** [comment]
@dimitropoulos hi! I'm looking to contribute for the first time here & this looks like a good candidate as you have pointed out. Is this still something I could look into fixing?

**Comment by dimitropoulos** [comment]
Definitely! Let me know if I can help!

**Comment by dopecodez** [comment]
I'd like to do take this up if its available. @parynaz , do you still plan on contributing to this issue?

**Comment by dimitropoulos** [comment]
thanks again @dopecodez for your contribution!
